peer/handler/MenuHandler.py

```diff
@@ -9,18 +14,210 @@ def serve(self, choice: str) -> None:
 		:param choice: the choice to handle
 		:return: None
 		"""
+		sp_ip4 = LocalData.get_superpeer_ip4()
+		sp_ip6 = LocalData.get_superpeer_ip6()
+		sp_port = LocalData.get_superpeer_port()
+		p_ssid = LocalData.session_id
 
 		if choice == "ADFF":
-			pass
+			shell.print_blue('\nFiles available for sharing:')
+			for count, file in enumerate(os.scandir('shared')):
+				file_md5 = hasher.get_md5(f'shared/{file.name}')
+				print(f'{count}] {file.name} | {file_md5}')
+			index = input('Choose a file to share (q to cancel): ')
+
+			if index == "q":
+				print('\n')
+				return
+
+			files = LocalData.get_shared_files()
+
+			while True:
+				try:
+					index = int(index) - 1
+				except ValueError:
+					shell.print_red(f'\nWrong index: number in range 1 - {len(files)} expected\n')
+					continue
+
+				if 0 <= index <= len(files)-1:
+					file_name = LocalData.get_shared_file_name(files[index])
+					file_md5 = LocalData.get_shared_file_md5(files[index])
+
+					# controlla se il file è già in condivisione
+					if not LocalData.is_shared_file(files[index]):
+
+						# crea il pacchetto e lo invia al super peer
+						packet = choice + p_ssid + file_md5 + file_name.ljust(100)
+						try:
+							net_utils.send_packet_and_close(sp_ip4, sp_ip6, sp_port, packet)
+							shell.print_green(f'Packet sent to supernode: {sp_ip4}|{sp_ip6} [{sp_port}]')
+						except net_utils.socket.error:
+							shell.print_red(f'\nError while sending packet to supernode: {sp_ip4}|{sp_ip6} [{sp_port}]\n')
+
+						# aggiunge il file alla struttura dati ed esce
+						LocalData.add_shared_file(file_md5, file_name)
+						shell.print_blue(f'\nNew shared file added {file.name} | {file_md5}')
+						break
+					else:
+						# il file è già in condivisione
+						shell.print_yellow(f'\n{file_name} | {file_md5} already in sharing.\n')
+						break
+				else:
+					shell.print_red(f'\nWrong index: number in range 1 - {len(files)} expected\n')
+					continue
 
 		elif choice == "DEFF":
-			pass
+			# scelta del file da rimuovere
+			shell.print_blue('File currently in sharing:')
+			for count, file in enumerate(LocalData.get_shared_files(), start=1):
+				print(f'{count}] {LocalData.get_shared_file_name(file)} | {LocalData.get_shared_file_md5(file)}')
+
+			while True:
+				index = input('Choose a file to remove from sharing (q to cancel): ')
+
+				if index == "q":
+					print('\n')
+					return
+
+				# recupera i files in sharing
+				files = LocalData.get_shared_files()
```

Comment:
Magari spostiamo questa riga di codice sopra al ciclo for, così non dobbiamo recuperarli due volte (in enumerate e qua)